fiftyone/utils/utils3d.py

```diff
@@ -0,0 +1,394 @@
+        and shading_mode != "height"
+    ):
+        if shading_mode == "rgb":
+            image_map[np.int_(points[:, 0]), np.int_(points[:, 1]), :] = (
+                colors * 255.0
+            )
+        else:
+            # use R channel for intensity, discard G and B channels
+            min_intensity = np.min(colors[:, 0])
+            max_intensity = np.max(colors[:, 1])
+            intensities_normalized_t = (colors[:, 0] - min_intensity) / (
+                max_intensity - min_intensity
+            )
+            # map intensity value to RGB
+            rgb_refs = _clamp_to_discrete(
+                intensities_normalized_t, list(shading_gradient_map.keys())
+            )
+            rgbs = np.array([shading_gradient_map[v] for v in rgb_refs])
+            image_map[np.int_(points[:, 0]), np.int_(points[:, 1]), :] = rgbs
+    elif shading_mode == "height":
+        # color by height (z)
+        max_z = np.max(points[:, 2])
+        min_z = np.min(points[:, 2])
+        z_normalized = (points[:, 2] - min_z) / (max_z - min_z)
+        # map z value to color
+        rgb_refs = _clamp_to_discrete(
+            z_normalized, list(shading_gradient_map.keys())
+        )
+        rgbs = np.array([shading_gradient_map[v] for v in rgb_refs])
+        image_map[np.int_(points[:, 0]), np.int_(points[:, 1]), :] = rgbs
+    else:
+        image_map[np.int_(points[:, 0]), np.int_(points[:, 1]), :] = 255.0
+
+    # change axis orientation such that y is up
+    image_map = np.rot90(image_map, k=1, axes=(0, 1))
+    bounds = [min_bound[0], max_bound[0], min_bound[1], max_bound[1]]
+    return image_map, bounds
+
+
+def _get_point_cloud_slice(samples):
+    point_cloud_slices = {
+        s for s, m in samples.group_media_types.items() if m == fom.POINT_CLOUD
+    }
+    if not point_cloud_slices:
+        raise ValueError("%s has no point cloud slices" % type(samples))
+
+    slice_name = next(iter(point_cloud_slices))
+
+    if len(point_cloud_slices) > 1:
+        logger.warning(
+            "Found multiple point cloud slices; using '%s'", slice_name
+        )
+
+    return slice_name
+
+
+def _clamp_to_discrete(arr: np.ndarray, discrete: list[float]):
+    """
+    Discretize by mapping each continuous value in ``arr``
+    to the closest value in ``discrete``.
+    """
+    clamp_list = np.sort(np.array(discrete))
+    idx = np.searchsorted(clamp_list, arr - 1e-8)
+    return clamp_list[np.clip(idx, 0, len(clamp_list) - 1)]
+
+
+# Reference: https://math.stackexchange.com/q/180418
+def _rotation_matrix_from_vectors(vec1, vec2):
+    """Returns the rotation matrix that aligns vec1 to vec2."""
+    a = (np.asarray(vec1) / np.linalg.norm(vec1)).reshape(3)
+    b = (np.asarray(vec2) / np.linalg.norm(vec2)).reshape(3)
+    v = np.cross(a, b)
+    c = np.dot(a, b)
+    s = np.linalg.norm(v)
+    K = np.array([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])
+    return np.eye(3) + K + K.dot(K) * ((1 - c) / (s**2))
```

Comment:
This will not be needed

Reply:
See previous comment. I think `projection_normal` is useful.